### ISSUE TITLE
Trilinos wrappers: Assert program is not distributed when using `operator=(dealii::Vector)` 

### ISSUE THREAD

**Issue by jpthiele** [dealii/dealii#17216]
I think the usage of `operator=` in the Trilinos wrappers to assign a non-distributed serial `dealii::Vector`
when using MPI with more than one process would lead to very strange behaviour and is not really advised.

So I was wondering whether we should not add an assertion that this is not the case 
to prevent possibly hard to track errors, especially for users new to distributed computing.

**Comment by bangerth** [comment]
I can see valid ways whereby one needs to create a serial representation and then uses the operator to distribute it to all processes. What problems do you foresee?

**Comment by jpthiele** [comment]
I agree that an assignment of values from a serial but globally indexed vector to a distributed one might be useful.
But if I see it correctly, the current Map/IndexSet is ignored during `operator=` and a purely local vector is constructed
and that behaviour might be unexpected.

For a copy constructor (which so far is missing) we definitely need to pass at least one IndexSet, then it should be fine too. That also is the way it was handled for Epetra.

**Comment by bangerth** [comment]
Yes, that (the fact that we use the map of the lhs vector) seems like a bug then. I'm entirely in favor of either fixing this, or disallowing it (via an assertion, or via the compiler).
